plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/index.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { recordEvent } from '@woocommerce/tracks';
+
+const actionButtons = document.querySelectorAll( '.row-actions span' );
+const bulkActions = document.querySelector( '#bulk-action-selector-top' );
+const bulkActionsButton = document.querySelector( '#doaction' );
+const featuredButtons = document.querySelectorAll( '#the-list .featured a' );
+const filterButton = document.querySelector( '#post-query-submit' );
+const productCategory = document.querySelector( '#product_cat' );
+const productType = document.querySelector( '#dropdown_product_type' );
+const searchButton = document.querySelector( '#search-submit' );
+const searchInput = document.querySelector( '#post-search-input' );
+const sortableColumnHeaders = document.querySelectorAll(
+	'.wp-list-table.posts thead .sortable a, .wp-list-table.posts thead .sorted a'
+);
+const stockStatus = document.querySelector( '[name="stock_status"]' );
+
+filterButton.addEventListener( 'click', function () {
+	recordEvent( 'products_list_filter_click', {
+		search_string_length: searchInput.value.length,
```

Comment:
I get an error here after I filtered by a product type that resulted in an empty list, I then reset the product type filter and hit `Filter` again.
In an empty list the `searchInput` isn't present, so it errors on `searchInput.value.length`.

<img width="1313" alt="Screen Shot 2022-05-16 at 2 01 05 PM" src="https://user-images.githubusercontent.com/2240960/168645484-b9c2ae12-5fcf-4bf0-b00e-46dbc7b32f47.png">

Reply:
👍